packages/main/src/Toolbar.ts

```diff
@@ -0,0 +1,590 @@
+import executeTemplate from "@ui5/webcomponents-base/dist/renderer/executeTemplate.js";
+import UI5Element from "@ui5/webcomponents-base/dist/UI5Element.js";
+import customElement from "@ui5/webcomponents-base/dist/decorators/customElement.js";
+import property from "@ui5/webcomponents-base/dist/decorators/property.js";
+import slot from "@ui5/webcomponents-base/dist/decorators/slot.js";
+import event from "@ui5/webcomponents-base/dist/decorators/event.js";
+import litRender from "@ui5/webcomponents-base/dist/renderer/LitRenderer.js";
+import { renderFinished } from "@ui5/webcomponents-base/dist/Render.js";
+import ResizeHandler from "@ui5/webcomponents-base/dist/delegate/ResizeHandler.js";
+import type { ResizeObserverCallback } from "@ui5/webcomponents-base/dist/delegate/ResizeHandler.js";
+import "@ui5/webcomponents-icons/dist/overflow.js";
+import arraysAreEqual from "@ui5/webcomponents-base/dist/util/arraysAreEqual.js";
+import Integer from "@ui5/webcomponents-base/dist/types/Integer.js";
+import Button from "./Button.js";
+
+import Popover from "./Popover.js";
+
+import ToolbarTemplate from "./generated/templates/ToolbarTemplate.lit.js";
+import ToolbarCss from "./generated/themes/Toolbar.css.js";
+
+import ToolbarPopoverTemplate from "./generated/templates/ToolbarPopoverTemplate.lit.js";
+import ToolbarPopoverCss from "./generated/themes/ToolbarPopover.css.js";
+
+import ToolbarDesign from "./types/ToolbarDesign.js";
+import ToolbarStyling from "./types/ToolbarStyling.js";
+import ToolbarAlign from "./types/ToolbarAlign.js";
+
+import ToolbarButton from "./ToolbarButton.js";
+import ToolbarSpacer from "./ToolbarSpacer.js";
+import ToolbarItem from "./ToolbarItem.js";
+import ToolbarSeparator from "./ToolbarSeparator.js";
+
+type ActionsWidthMap = {
+	[key: string]: number
+};
+
+/**
+ * @class
+ *
+ * <h3 class="comment-api-title">Overview</h3>
+ *
+ * The <code>ui5-toolbar</code> component is used to create a horizontal layout with actions.
+ * The items can be overflowing in a popover, when the space is not enough to show all of them.
+ *
+ * <h3>Keyboard Handling</h3>
+ * The <code>ui5-toolbar</code> provides advanced keyboard handling.
+ * <br>
+ * <ul>
+ * <li>The control is not interactive, but can contain of interactive elements </li>
+ * <li>[TAB] - iterates through elements</li>
+ * </ul>
+ * <br>
+ *
+ * <h3>ES6 Module Import</h3>
+ * <code>import "@ui5/webcomponents/dist/Toolbar";</code>
+ * @constructor
+ * @author SAP SE
+ * @alias sap.ui.webc.main.Toolbar
+ * @extends sap.ui.webc.base.UI5Element
+ * @tagname ui5-toolbar
+ * @public
+ * @since 1.16.0
+ */
+@customElement({
+	tag: "ui5-toolbar",
+	languageAware: true,
+	renderer: litRender,
+	template: ToolbarTemplate,
+	staticAreaTemplate: ToolbarPopoverTemplate,
+	styles: ToolbarCss,
+	staticAreaStyles: ToolbarPopoverCss,
+	dependencies: [
+		Popover,
+		Button,
+		ToolbarButton,
```

Comment:
Why does it have a strict dependency to specific items? I thought the idea was to not depend on any one item (only know about the abstract item interface) and work with any item that is passed to it, provided the item satisfies the interface